src/main/java/pl/sda/library/infrastructure/jpa/AuthorRepositoryAdapter.java

```diff
@@ -16,36 +19,46 @@
 
     @Override
     public Long addAuthor(Author author) {
-        return null;
+        if(isExistByFirstNameAndLastName(author.getFirstName(), author.getLastName())){
+            throw new AuthorAlreadyExistException();
+        }
+        return authorDtoRepository.save(AuthorDto.fromDomain(author)).getId();
     }
 
     @Override
     public Optional<Author> getAuthorById(Long id) {
+        Optional<AuthorDto> byId = authorDtoRepository.findById(id);
```

Comment:
cala metoda moze miec jedna linijke:
return authorDtoRepository.findById(id).map(authorDto -> authorDto.toModel())

Reply:
> cala metoda moze miec jedna linijke:
> return authorDtoRepository.findById(id).map(authorDto -> authorDto.toModel())

a nie będzie jakiś wyjątków jak findById nic nie znajdzie?

zdążyłem już zacomitować takie rozwiązanie
@Override
    public Optional<Author> getById(Long id) {
        Optional<AuthorDto> byId = authorDtoRepository.findById(id);
        if(byId.isPresent()){
            return Optional.of(byId.get().toModel());
        }
        return Optional.empty();
    }
Może być?

Reply:
to wlasnie ogarnia Optional ze nie bedzie
map sie zadzieje tylko jesli obiekt w optionalu istnieje